File: cases/taint/php/dispatch-table-positive/dispatch_table_positive.php
<?php
function dfb_source(): string { // DFB-SOURCE: dispatch-table-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: dispatch-table-sink

function run(): void {
    $table = [ // DFB-WITNESS: dispatch-table-build
        "leak" => function (string $value): void {
            dfb_sink($value);
        },
        "drop" => function (string $value): void {
            dfb_sink("clean");
        },
    ];
    $key = "leak";
    $table[$key](dfb_source());
}

File: cases/taint/cpp/dispatch-table-positive/dispatch_table_positive.cpp
#include <functional>
#include <map>
#include <string>

std::string dfb_source() { // DFB-SOURCE: dispatch-table-input
    return "tainted";
}

void dfb_sink(const std::string &value) {} // DFB-SINK: dispatch-table-sink

void leak(const std::string &value) {
    dfb_sink(value);
}

void drop(const std::string &value) {
    dfb_sink("clean");
}

void run() {
    std::map<std::string, std::function<void(const std::string &)>> table; // DFB-WITNESS: dispatch-table-build
    table["leak"] = leak;
    table["drop"] = drop;
    std::string key = "leak";
    table[key](dfb_source());
}
